app/components/tree_link_component.rb

```diff
@@ -2,7 +2,7 @@
 
 class TreeLinkComponent < ViewComponent::Base
   include MultiLanguagesHelper, ModalHelper, ApplicationHelper
-  def initialize(child:, href:, children_href: , selected: false , data: {}, muted: false, target_frame: nil, open_in_modal: false)
+  def initialize(child:, href:, children_href: , selected: false , data: {}, muted: false, target_frame: nil, open_in_modal: false, ontology_uri_pattern: nil)
```

Comment:
put here as parameter `is_reused: false` instead of sending the URI.

Reply:
We can't know whether it is reused or not in this level, I need to compare the ontoloy_uri_pattern with every concept's id to know whether it's reused (it has an id that doesn't follow the ontology uri pattern) or it is not reused (it has that pattern in the id)

Reply:
yes, you put the same logic, but outside of this component, when you call it you do the logic and pass `is_reused; true` or `false`
instead of passing  `ontology_uri_pattern` you pass directly the result of your logic.